ethclient/ethclient.go

```diff
@@ -505,6 +505,46 @@ func (ec *Client) SuggestGasTipCap(ctx context.Context) (*big.Int, error) {
 	return (*big.Int)(&hex), nil
 }
 
+// FeeHistoryResult is the result of FeeHistory.
+type FeeHistoryResult struct {
+	OldestBlock  *big.Int     `json:"oldestBlock"`
+	Reward       [][]*big.Int `json:"reward,omitempty"`
+	BaseFee      []*big.Int   `json:"baseFeePerGas,omitempty"`
+	GasUsedRatio []float64    `json:"gasUsedRatio"`
+}
+
+type feeHistoryResultMarshaling struct {
+	OldestBlock  *hexutil.Big     `json:"oldestBlock"`
+	Reward       [][]*hexutil.Big `json:"reward,omitempty"`
+	BaseFee      []*hexutil.Big   `json:"baseFeePerGas,omitempty"`
+	GasUsedRatio []float64        `json:"gasUsedRatio"`
+}
+
+// FeeHistory retrieves the fee market history.
+func (ec *Client) FeeHistory(ctx context.Context, blockCount uint64, lastBlock *big.Int, rewardPercentiles []float64) (*FeeHistoryResult, error) {
+	var res feeHistoryResultMarshaling
+	if err := ec.c.CallContext(ctx, &res, "eth_feeHistory", hexutil.Uint(blockCount), hexutil.Big(*lastBlock), rewardPercentiles); err != nil {
```

Comment:
You have to encode the block number with `toBlockNumArg`

Reply:
Good point - should be resolved now. Thanks!